### PR TITLE
docs(pubsub): add doxygen comment to deprecate handwritten apis

### PR DESCRIPTION
**Pull request by alevenberg**
Part of https://github.com/googleapis/google-cloud-cpp/issues/12987

This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13585)

### CODE REVIEW

File: google/cloud/pubsub/subscription_admin_client.h [hunk 1]
```diff
@@ -31,6 +31,9 @@ GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 /**
  * Performs subscription administrative operations in Cloud Pub/Sub.
  *
+ * @deprecated Please use #google::cloud::pubsub_admin::SubscriptionAdminClient
+ *     and #google::cloud::pubsub_admin::SubscriptionAdminClient instead.
```
Comment (coryan) [inline google/cloud/pubsub/subscription_admin_client.h:35]: The Doxygen documentation recommends `\ref` (aka `@ref`) to link classes:

https://www.doxygen.nl/manual/autolink.html

Can you run the `publish-docs-pr` build locally and see if the generated `HTML` files look right to you?

**Review by coryan — APPROVED** [review]

File: google/cloud/pubsub/subscription_admin_client.h (hunk 1 above)
Comment (alevenberg) [reply google/cloud/pubsub/subscription_admin_client.h]: Ack, will run locally

Comment (alevenberg) [reply google/cloud/pubsub/subscription_admin_client.h]: This is what it looks like  (which I think is fine)

```
<div class="line"><a id="l00031" name="l00031"></a><span class="lineno">   31</span><span class="comment">/**</span></div>
<div class="line"><span class="lineno">   32</span><span class="comment"> * Performs subscription administrative operations in Cloud Pub/Sub.</span></div>
<div class="line"><span class="lineno">   33</span><span class="comment"> *</span></div>
<div class="line"><span class="lineno">   34</span><span class="comment"> * @deprecated Please use \ref</span></div>
<div class="line"><span class="lineno">   35</span><span class="comment"> * google::cloud::pubsub_admin::SubscriptionAdminClient and \ref</span></div>
<div class="line"><span class="lineno">   36</span><span class="comment"> * google::cloud::pubsub_admin::SubscriptionAdminClient instead.</span></div>
<div class="line"><span class="lineno">   37</span><span class="comment"> *</span></div>
<div class="line"><span class="lineno">   38</span><span class="comment"> * Applications use this class to perform subscription administrative operations</span></div>
<div class="line"><span class="lineno">   39</span><span class="comment"> * on [Cloud Pub/Sub][pubsub-doc-link].</span></div>
```

Comment (dbolduc) [reply google/cloud/pubsub/subscription_admin_client.h]: I feel like they should be hyperlinked with `<a>` tags. What does the html look like rendered in the browser?

File: google/cloud/pubsub/subscription_admin_connection.h [hunk 2]
```diff
@@ -63,6 +63,10 @@ using ListSnapshotsRange =
  * A connection to Cloud Pub/Sub for subscription-related administrative
  * operations.
  *
+ * @deprecated Please use \ref
+ * google::cloud::pubsub_admin::SubscriptionAdminClient and \ref
+ * google::cloud::pubsub_admin::SubscriptionAdminClient instead.
```
Comment (dbolduc) [inline google/cloud/pubsub/subscription_admin_connection.h:68]: s/Client/Connection/

File: google/cloud/pubsub/topic_admin_connection.h [hunk 3]
```diff
@@ -73,6 +73,9 @@ using ListTopicSnapshotsRange =
 /**
  * A connection to Cloud Pub/Sub for topic-related administrative operations.
  *
+ * @deprecated Please use #google::cloud::pubsub_admin::SubscriptionAdminClient
+ *     and #google::cloud::pubsub_admin::SubscriptionAdminClient instead.
```
Comment (dbolduc) [inline google/cloud/pubsub/topic_admin_connection.h:77]: s/Subscription/Topic/

s/Client/Connection/

**Pull request merged**